fix: Infinite NetworkManager.ShutdownInternal on NetworkTransport.DisconnectLocalClient Exception [MTT-5551]

NoelStephensUnity opened the pull request:
This PR just adds two try catch wrappers to handle two scenarios that could disrupt the shutdown/disconnect sequence.
NetworkManager: The try catch wrapper assures that it will continue to process the remainder of the ShutdownInternal code if NetworkTransport.DisconnectLocalClient throws and exception (_otherwise it will end up in an infinite shutdown sequence_).

UnityTransport: The try catch wrapper assures the invoking methods will continue to process if FlushSendQueuesForClientId throws an exception.

[MTT-5551](https://jira.unity3d.com/browse/MTT-5551)

## Testing and Documentation

- No tests have been added.
- No documentation changes or additions were necessary.

NoelStephensUnity committed:
fix

Prevent an infinite shutdown sequence from happening if NetworkTransport.DisconnectLocalClient throws an exception during NetworkManager.ShutdownInternal.

NoelStephensUnity committed:
fix

If UnityTransport.FlushSendQueuesForClientId throws an exception, log the exception and allow the invoking method to continue processing.

NoelStephensUnity committed:
style

Fixing white space and removing extra LF

simon-lemay-unity commented on com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs:
```diff
@@ -1085,13 +1085,22 @@ private void ClearSendQueuesForClientId(ulong clientId)
-            foreach (var kvp in m_SendQueue)
+            // In the event this throws and exception, log it
+            // and allow the invoking method to continue.
+            try
```
I don't think catching the exception is the correct course of action here. We should simply check `m_Driver.IsCreated` (perhaps directly in `SendBatchedMessages`) and exit early if it's not created. The problem with the exception is that it will only be thrown when safety checks are enabled (i.e. in the editor). In a player no exception will be thrown and instead we'll end up accessing garbage memory (with failure modes that are less predictable).

NoelStephensUnity replied:
Thanks for catching that! 👍 
Made the changes.

NoelStephensUnity committed:
update

Removing try catch from UnityTransport and replacing that with a check if the driver is created before trying to send batched messages.

Migrating the unregistering from NetworkUpdateLoop to occur above where it could potentially disconnect the client.

NoelStephensUnity committed:
style

remove comment that is no longer needed.

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs:
```diff
@@ -1197,10 +1197,22 @@ internal void ShutdownInternal()
+            // Unregister network updates before trying to disconnect the client
+            this.UnregisterAllNetworkUpdates();
+
             if (IsClient && IsListening)
```
Wondering if this check needs to be:
`if (!IsServer && IsClient && IsListening)`
Or... the comment below needs to be updated.
